<a href="https://colab.research.google.com/github/19PA1A05I2/VITBML_EDA1/blob/master/AI_LEARNERS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**IRIS DATA SET VISUALISATION**

In [0]:
# Importing the libraries
import numpy as np 
import pandas as pd

In [0]:
dataset = pd.read_csv('/content/iris-flower-dataset.zip')
print(dataset.columns)
dataset.head(3)

Index(['sepal_length', 'sepal_width', 'petal_length', 'petal_width',
       'species'],
      dtype='object')


sepal_length  sepal_width  petal_length  petal_width      species
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa

In [0]:
# Importing metrics for evaluation
from sklearn.metrics import confusion_matrix
from sklearn.metrics import classification_report

In [0]:
from sklearn.model_selection import train_test_split
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state =452 )

**Logistic Regression** 

Confusion Matrix

Accuracy

In [0]:
#importing libraries
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression()
classifier.fit(X_train, y_train)

y_pred = classifier.predict(X_test)

# Summary of the predictions made by the classifier
print(classification_report(y_test, y_pred))
print(confusion_matrix(y_test, y_pred))
# Accuracy score
from sklearn.metrics import accuracy_score
print('accuracy is',accuracy_score(y_pred,y_test))


                 precision    recall  f1-score   support

    Iris-setosa       1.00      1.00      1.00        10
Iris-versicolor       0.92      1.00      0.96        11
 Iris-virginica       1.00      0.89      0.94         9

       accuracy                           0.97        30
      macro avg       0.97      0.96      0.97        30
   weighted avg       0.97      0.97      0.97        30

[[10  0  0]
 [ 0 11  0]
 [ 0  1  8]]
accuracy is 0.9666666666666667


**Support Vector Machines(SVM)**

Confusion Matrix

Accuracy

In [0]:
from sklearn.svm import SVC

classifier = SVC()
classifier.fit(X_train, y_train)

y_pred = classifier.predict(X_test)

# Summary of the predictions made by the classifier
print(classification_report(y_test, y_pred))
print(confusion_matrix(y_test, y_pred))
# Accuracy score
from sklearn.metrics import accuracy_score
print('accuracy is',accuracy_score(y_pred,y_test))

                 precision    recall  f1-score   support

    Iris-setosa       1.00      1.00      1.00        10
Iris-versicolor       1.00      1.00      1.00        11
 Iris-virginica       1.00      1.00      1.00         9

       accuracy                           1.00        30
      macro avg       1.00      1.00      1.00        30
   weighted avg       1.00      1.00      1.00        30

[[10  0  0]
 [ 0 11  0]
 [ 0  0  9]]
accuracy is 1.0


**Decision Tree**

Confusion Matrix

Accuracy

In [0]:
from sklearn.tree import DecisionTreeClassifier

classifier = DecisionTreeClassifier()

classifier.fit(X_train, y_train)

y_pred = classifier.predict(X_test)

# Summary of the predictions made by the classifier
print(classification_report(y_test, y_pred))
print(confusion_matrix(y_test, y_pred))
# Accuracy score
from sklearn.metrics import accuracy_score
print('accuracy is',accuracy_score(y_pred,y_test))

                 precision    recall  f1-score   support

    Iris-setosa       1.00      1.00      1.00        10
Iris-versicolor       0.91      0.91      0.91        11
 Iris-virginica       0.89      0.89      0.89         9

       accuracy                           0.93        30
      macro avg       0.93      0.93      0.93        30
   weighted avg       0.93      0.93      0.93        30

[[10  0  0]
 [ 0 10  1]
 [ 0  1  8]]
accuracy is 0.9333333333333333


**K Nearest Neighbours(KNN)**

Confusion Matrix

Accuracy

In [0]:
from sklearn.neighbors import KNeighborsClassifier  #used to build knn model
from sklearn import metrics                         #used to compute accuracy of the model
import sklearn.model_selection as ms     
y=dataset['species']    #assigns the 'species' attribute as the target variable
x = dataset.drop("species",axis=1) #x contains all the predictors of y.hence the 'id' and 'species' columns are dropped from iris

x_train,x_test,y_train,y_test=ms.train_test_split(x,y,test_size=0.3,random_state=1) #the dataset is split into train and test data

#build the knn model for k=7
knn1 = KNeighborsClassifier(n_neighbors=7, metric='euclidean')
knn1.fit(x_train,y_train)
y_pred = knn1.predict(x_test)
print("The accuracy of the model: ",metrics.accuracy_score(y_test,y_pred))

#To predict the species of 2 unknown iris flowers
unknown = [[76,34,5,2.4],       
           [5,3.9,2,0.3]] #assigning the characteristics of 2 iris flowers of unknown species to unknown
pred = knn1.predict(unknown)  #predicting the species of the unknown iris flowers
print("Predicted species: ", pred)   #print the predicted species


The accuracy of the model:  0.9777777777777777
Predicted species:  ['Iris-virginica' 'Iris-setosa']


**CONCLUSION**

*We have taken the data of iris-flower.*

The accuracy in each method is as follows :

Logistic Regression - 96.6%

SVM - 100%

DECISION TREE - 93.3%

KNN - 97.7%

SVM Technique worked best for the data since it has the good accuracy.


# TEAM CONTRIBUTORS

**U.PRIYANKA - 19PA1A05I2**

**SK.ARSHIYA FARHEEN JAHAN - 19PA1A05G7**

**P.MONICA RANI - 19PA1A05E1**

**D.VAISHNAVI - 19PA1A0538**